src/System.CommandLine/Invocation/TypoCorrection.cs

```diff
@@ -23,9 +23,10 @@ public TypoCorrection(int maxLevenshteinDistance)
 
         public void ProvideSuggestions(ParseResult result, IConsole console)
         {
-            for (var i = 0; i < result.UnmatchedTokens.Count; i++)
+            var unmatchedTokens = result.UnmatchedTokens;
```

Comment:
we need to store the result, because currently it's always allocating: https://github.com/dotnet/command-line-api/blob/905cb0b525a97ba753026954b4551fecd28375bd/src/System.CommandLine/ParseResult.cs#L118